cdap-data-fabric/src/main/java/co/cask/cdap/data2/dataset2/lib/partitioned/PartitionedFileSetDataset.java

```diff
@@ -152,6 +154,19 @@ public PartitionedFileSetDataset(DatasetContext datasetContext, String name,
   }
 
   @Override
+  public void setContext(RuntimeProgramContext context) {
+    this.runtimeProgramContext = context;
+  }
+
+  /**
+   * Returns the current {@link RuntimeProgramContext} or {@code null} if it is not available.
+   */
+  @Nullable
+  protected RuntimeProgramContext getRuntimeProgramContext() {
```

Comment:
It was actually intended to be protected so that sub-class (e.g. `TimePartitionedFileSetDataset`) can access the context if needed (since this class implements context aware, so all sub-classes should be able to get the context).